### DISCUSSION TITLE
API related questions

### DISCUSSION THREAD

**Discussion by Fe1son** [TownyAdvanced/Towny#6127] [Q&A] [answered]
Two questions... 

1. Is there event to listen unclaim townblock?

2. Is there event to listen when player claims a new plot or when player sets plot for sale?

**Comment by LlmDl** [comment, marked as answer]
There are lots of events:
https://github.com/TownyAdvanced/Towny/tree/master/src/com/palmergames/bukkit/towny/event

You're asking about the:
- [TownUnclaimEvent.java](https://github.com/TownyAdvanced/Towny/blob/master/src/com/palmergames/bukkit/towny/event/town/TownUnclaimEvent.java)
- [PlotClaimEvent.java](https://github.com/TownyAdvanced/Towny/blob/master/src/com/palmergames/bukkit/towny/event/plot/changeowner/PlotClaimEvent.java)
- [PlotSetForSaleEvent.java](https://github.com/TownyAdvanced/Towny/blob/master/src/com/palmergames/bukkit/towny/event/plot/PlotSetForSaleEvent.java)
